testing/regress/ecl/spray_replicate_test.ecl

```diff
@@ -0,0 +1,149 @@
+/*##############################################################################
+
+    Copyright (C) 2018 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+//noHThor
+
+//class=spray
+
+import std.system.thorlib;
+import Std.File AS FileServices;
+
+jlib:= SERVICE
+    unsigned8 rtlTick() : library='jlib',eclrtl,entrypoint='rtlNano';
+END;
+
+dropzonePath := '/var/lib/HPCCSystems/mydropzone/' : STORED('dropzonePath');
+espIpPort := 'http://127.0.0.1:8010/FileSpray' : STORED('espIpPort');
+engine := thorlib.platform();
+prefix := engine + '-';
+suffix := '-' + jlib.rtlTick() : STORED('timeStamp');
+
+sprayPrepFileName := '~spray_prep';
+desprayOutFileName := dropzonePath + 'file_for_spray' + suffix;
+sprayOutFileName := prefix + 'spray_wo_replication' + suffix;
+
+unsigned VERBOSE := 0;
+
+Layout_Person := RECORD
+  STRING3  name;
+  UNSIGNED2 age;
+  BOOLEAN good;
+END;
+
+empty := DATASET([], Layout_Person);
+
+dsSetup := DATASET([ {'foo', 10, 1},
+                     {'bar', 12, 0},
+                     {'baz', 32, 1} ]
+            ,Layout_Person);
+
+//  Create a small logical file
+setup := output(dsSetup, , sprayPrepFileName, CSV, OVERWRITE);
+
+rec := RECORD
+  string result;
+  string msg;
+end;
+
+
+// Despray it to default drop zone
+rec despray(rec l) := TRANSFORM
+  SELF.msg := FileServices.fDespray(
+                       LOGICALNAME := sprayPrepFileName
+                      ,DESTINATIONIP := '.'
+                      ,DESTINATIONPATH := desprayOutFileName
+                      ,ALLOWOVERWRITE := True
+                      );
+  SELF.result := 'Despray Pass';
+end;
+
+dst1 := NOFOLD(DATASET([{'', ''}], rec));
+p1 := NOTHOR(PROJECT(NOFOLD(dst1), despray(LEFT)));
+c1 := CATCH(NOFOLD(p1), ONFAIL(TRANSFORM(rec,
+                                 SELF.result := 'Despray Fail',
+                                 SELF.msg := FAILMESSAGE
+                                )));
+#if (VERBOSE = 1)
+    desprayOut := output(c1);
+#else
+    desprayOut := output(c1, {result});
+#end
+
+rec spray(rec l) := TRANSFORM
+    SELF.msg := FileServices.fSprayDelimited(
+                        SOURCEIP := '.',
+                        SOURCEPATH := desprayOutFileName,
+                        DESTINATIONGROUP := 'my' + engine,
```

Comment:
This won't work in general, e.g. if environment is not completely standard.

Using Std.System.Thorlib.group() should work I think.